### ISSUE TITLE
Changing the name of the torrent

### ISSUE THREAD

**Issue by peaveyman** [transmission-remote-gui/transgui#919]
This just started happening a couple weeks ago. When I add a torrent and change anything in the title of the torrent, for example, I change [FLAC] to [Flac], the cursor jumps forward or back depending on if I'm backspacing or hitting the delete key to remove the LAC. It also does this if I hit F2 to rename a torrent that's already finished downloading. I had this issue on Plasma 5 and on Gnome.

**Comment by peaveyman** [comment]
Has anybody seen this problem?

**Comment by Gittyperson** [comment]
I'm also unable to rename torrents ("input/output error" message appears) and wanted to ask about it. This could be a Transmission limitation. By _torrent name_ I only mean the way it appears on the UI list, nothing else.

For example, with a torrent that has a parent folder, it seems the _torrent name_  always follows the name of that parent folder. If I want use another folder or relocate the torrent's contents, the torrent's name will also have to change. I'd be great if we could do without this limitation.

I am not talking about renaming files/folders within the torrent's contents, this example is only about the parent folder. Haven't tested yet if torrent renaming is possible when there is no parent folder.

**Comment by peaveyman** [comment]
I have no problem renaming a torrent. It's just that the cursor jumps places when I hit the backspace or delete key.

**Comment by peaveyman** [comment]
Any ideas on this problem?

**Comment by Davikar** [comment]
I have the same problem. The cursor jumps in the wrong direction after using backspace and delete. This is on Manjaro Linux with both KDE and Gnome. Works fine on Windows.

**Comment by peaveyman** [comment]
I'm on Manjaro also. KDE and Gnome. Could this have something to do with it?

**Comment by Davikar** [comment]
Dunno. Manjaro/Arch and many other distros use newer versions of most programs than for example Ubuntu and Debian. Could be that the newer versions are less compatible with Transmission Remote GUI.

**Comment by peaveyman** [comment]
I can confirm this bug still exists on 5.5.1

**Comment by peaveyman** [comment]
I didn't mean to close this. Reopening it.

**Comment by Davikar** [comment]
Yes, I still have this problem on 5.5.1 too.

**Comment by leonsoft-kras** [comment]
Probably all have to guess that this problem is not in Windows.

This may well be a problem of implementation of the compiler and OS.
I did not change this part of the code, and in Windows function is working properly.
I will not help with Linux.

**Comment by leonsoft-kras** [comment]
old TRG compiled Lazarus 1.2.6
My TRG 5.5.1 - Lazarus 1.6.2 
My other versions TRG - Lazarus 1.2.6, 1.4.4, 1.6.0.

**Comment by Davikar** [comment]
Yes, it's working fine under Windows and when running it on Wine. Not sure if it worked fine with the old 5.0.1, I haven't tested.
It's not a big problem imho, but it's a bit annoying.

**Comment by peaveyman** [comment]
This had been working on release 5.7 but is back on 5.8

**Comment by leonsoft-kras** [comment]
Operating system?

The version has changed:
Lazarus 1.6.4 - fpc 3.0.2

**Comment by leonsoft-kras** [comment]
In which operating system is the problem?

**Comment by peaveyman** [comment]
Manjaro Linux.

**Comment by leonsoft-kras** [comment]
[transgui-5.8-i386-linux.zip](https://github.com/leonsoft-kras/transmisson-remote-gui/files/1133804/transgui-5.8-i386-linux.zip)

Please try. Report the result

**Comment by peaveyman** [comment]
I need the x86_64 zip.

**Comment by leonsoft-kras** [comment]
Probably someone will compile, as earlier (not through Travis CI)

**Comment by peaveyman** [comment]
I unzipped the download you linked to and ran it without installing. The problem still exists.

**Comment by peaveyman** [comment]
What changed between 5.7 and 5.8 to cause the regression?

**Comment by leonsoft-kras** [comment]
Nothing.

I can assume that the case in the lazarus and fpc.
I don't know which version of the compiler under Linux was used in version 5.7.
You can compare the sources of version 5.7 and 5.8. Change only in the SSL part.

**Comment by peaveyman** [comment]
The versions on my computer didn't change between 5.7 and 5.8

**Comment by alfonskunk** [comment]
The problem stills persist on 5.8.2 on ubuntu x64

**Comment by PeterDaveHello** [comment]
I can try to build it on an older Ubuntu like 12.04 or 14.04 and see if it works.

**Comment by PeterDaveHello** [comment]
I try to build it with a newer environment, can somebody please help test it?

1. https://file.io/rowyzj (built with Ubuntu 17.04, md5: `19c6f9ebd584b72e4e7010bfe51a4746`)
2. https://file.io/sjQooF (built with Ubuntu 17.10, md5: `708dd49801557c9a562b23c14fd7c489`)

**Comment by alfonskunk** [comment]
Hi Peter. Both of them works ok on Ubuntu 17.10 Alpha x64 and on Ubuntu 17.04 x64 too

Thanks!

**Comment by PeterDaveHello** [comment]
Thanks @alfonskunk, I'll change our building environment to Ubuntu 17.04 (as 17.10 is still under alpha), and plan another minor release recently, please let me know that if anything else got wrong after that. cc @leonsoft-kras

**Comment by PeterDaveHello** [comment]
These are the automatic builds on Travis CI, they'll be removed after one week, it'll be good if anyone wants to help test them or use if before the new release:
- Linux amd64: https://file.io/RFHwmb (md5: `51348e063bc407e2d851e6f64af23371`)
- Linux i386: https://file.io/Qarrjs (md5: `f440f3fdb5dff86b4a30a461dadd2cae`)
- Linux armv6l(Raspberry Pi): https://file.io/TbBVa4 (md5: `db439290e729e90327ebced50abce544`)
- macOS: https://file.io/Zy1bSc (md5: `229d99d1ce9580a04b9b26371eb8261f`)

Please let me know if there is anything else should be fixed in this issue.

For more info about this automatic build, or more checksums like sha1 or sha256, please take a look at: https://travis-ci.org/leonsoft-kras/transmisson-remote-gui/builds/258096347

**Comment by PeterDaveHello** [comment]
v5.8.3 released here: https://github.com/leonsoft-kras/transmisson-remote-gui/releases/tag/v5.8.3

**Comment by peaveyman** [comment]
I need a zip build so I can test this on Manjaro.

**Comment by peaveyman** [comment]
This is NOT fixed in 5.8.3. It is fixed in 5.7.1

**Comment by PeterDaveHello** [comment]
Use `txz` instead of `zip` please.

**Comment by peaveyman** [comment]
It's still not fixed.

**Comment by peaveyman** [comment]
I can confirm it's fixed in the newest release.

**Comment by peaveyman** [comment]
This has regressed in the new release 5.12

**Comment by PeterDaveHello** [comment]
hmmm, I doubt it's related with #884, cc @antekgla

**Comment by antekgla** [comment]
Don´t should be because #884 refer to context menu and happens in windows too... be who knows?
I dont have a linux machine to test this.
Maybe @PeterDaveHello can test removing this changes and make a linux compile to @peaveyman to test.

**Comment by PeterDaveHello** [comment]
@peaveyman try this one? https://file.io/ptqONh

**Comment by peaveyman** [comment]
That did not fix it.

**Comment by PeterDaveHello** [comment]
Okay, so it's not related.

**Comment by peaveyman** [comment]
Correct.

**Comment by peaveyman** [comment]
I can confirm this bug is still present in the new release.

**Comment by peaveyman** [comment]
Still no idea why this is happening?

**Comment by leonsoft-kras** [comment]
I think we should involve Lazarus developers. The source code for the program should work the same in any system. If in Windows there is no problem, and in Linux there are problems, it is a question to Lazarus and the compiler.

I write a lot of C/C++ programs using Qt Framework. I never had to use the "if linux then ... else ..." constructs. The whole difference in the operation of operating systems is hiding the Framework.

This post by the way concerns many other issues about which users of the program write.

**Comment by peaveyman** [comment]
Something else I've discovered. I run Manjaro linux which is based on Arch linux and can use the AUR. When I install this package, https://aur.archlinux.org/packages/transmission-remote-gui-bin/ it shows the older version of Lazarus and the renaming problem is there. When I install this package, https://aur.archlinux.org/packages/transgui-qt/ it shows the new version of Lazarus and the renaming problem is not there.

**Comment by leonsoft-kras** [comment]
what is written in "about"  each program?

**Comment by peaveyman** [comment]
This is transmission-remote-gui-bin: https://i.imgur.com/YSTtxav.png

This is transgui-qt: https://i.imgur.com/zFePqEI.png

**Comment by leonsoft-kras** [comment]
Well, what can I say - free open sources :)
If the source code didn't change in each case, then the difference in ...

**Comment by alfonskunk** [comment]
The difference is the building platform used. The PeterDaveHello from 07/27

> PeterDaveHello commented on 27 Jul 2017
I try to build it with a newer environment, can somebody please help test it?

Both of them works as expected.

**Comment by leonsoft-kras** [comment]
What is this message about?

**Comment by alfonskunk** [comment]
About the renaming problem in Linux binaries. On July (5.8.2 version), the problem solves with the same source code but different builiding environment, as PeterDaveHello wrotes then.

**Comment by PeterDaveHello** [comment]
Not sure if this problem still exists?

**Comment by Davikar** [comment]
No, not as long as you build it with the latest Lazarus version.

**Comment by rd1** [comment]
still happenning in v5.15.4

i just tried to rename a freshly added torrent, with 0% downloaded, and i still get this input/output error msgbox.

**Comment by Davikar** [comment]
@rd1 What version of Lazarus is your TR GUI compiled with? Mine is 1.8.2.0.

**Comment by Annihilannic** [comment]
I also have this problem under CentOS 7 Transmission Remote GUI 5.15.4 Lazarus 1.6.2.0 (the precompiled binaries provided on this site).

It's almost impossible to rename files or torrents because the cursor position does not behave predictably, moving forward and backward characters while you type.  It's almost like it doesn't know the character widths and keeps repositioning itself where it thinks it should be after each character is typed, but gets it wrong.

**Comment by leonsoft-kras** [comment]
I don't think this is a problem of the TRGUI because such small things should not be part of the program code. You should try using the newer versions of Lazarus and FPC.

**Comment by Annihilannic** [comment]
I did that, and now it works perfectly (FPC 3.0.4, Lazarus 1.8.4.0).  Thanks!

**Comment by rd1** [comment]
Davikar, sorry i didn't know you had replied.

i'm, using TRG 5.15.4 , i don't have a compiling environment, i download the binaries from here.
but TRG's About box says FPC 3.0.4, Lazarus 1.8.2.0

**Comment by Annihilannic** [comment]
I noticed a similar thing @rd1, my guess was because it's trying to rename a file that it hasn't created yet.  It's probably a problem with transmission rather than the GUI... in any case, I don't think it belongs to the original issue being discussed here.

**Comment by damojms** [comment]
Reporting this issue still present in 5.16.0 under GTK 3.24.4
I suspect it's either an issue in one of the TVarGrid edit functions or with Lazarus' underlying implementation of the string editor for GTK as I have also used the same version under windows and do not experience this issue.
This occurs both in rename torrent and rename file within a torrent.

**Comment by Annihilannic** [comment]
> Reporting this issue still present in 5.16.0 under GTK 3.24.4
> I suspect it's either an issue in one of the TVarGrid edit functions or with Lazarus' underlying ...

It was already mentioned in recent comments that this behaviour seems to relate to the Lazarus version - you didn't say which version you used to compile?

**Comment by damojms** [comment]
Yes, apologies. Lazarus version 1.8.4 & FPC 3.0.4

As a test I compiled this against Qt5 bindings and the issue is not present with this configuration.

I was also incorrect as the version I had was compiled against GTK2 bindings, not GTK3
